### PR TITLE
(web-components) Use `ElementInternals` for TextInput elements

### PR DESCRIPTION
**Pull request by radium-v**
## Previous Behavior

Currently, form-associated custom elements use the `FormAssociated` class mixin. This mixin is heavy and duplicative, being copied wholesale for every component which utilizes it. While `FormAssociated` includes minimal support for [`ElementInternals`](https://developer.mozilla.org/en-US/docs/Web/API/ElementInternals), the implementation isn't fully baked.

## New Behavior

Improvements:
- Switches the `<fluent-text-input>` custom element to use `ElementInternals` by default. An optional polyfill may be added in the near future, as more components transition to use ElementInternals.
- Adds additional tests for the `TextInput` component.
- Expands the properties and attributes for the Text Input storybook fixtures.

### CODE REVIEW

**Comment by chrisdholt** [comment]
Should we pull the label from within the control with this change?

**Review by janechu — APPROVED** [review]

**Comment by chrisdholt** [comment]
This will need to be pointed to `master` as we've merged the v3 branch :)

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: change/@fluentui-web-components-7fcf0c62-7015-4e62-b867-951d528a3a4c.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-web-components-7fcf0c62-7015-4e62-b867-951d528a3a4c.json:1]: ## [🕵 fluentui-web-components-v3 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31201&commitId=86b97b2bd60593c29985db5b7c443a97e4780669&env=prod&clientType=FLUENTUI)


  <details><summary><b>Accordion</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Accordion.Size - RTL.normal.chromium.png|2| Changed|
|Accordion.Size - Dark Mode.normal.chromium.png|2| Changed|
|Accordion.Size.normal.chromium.png|2| Changed|

</details>


  <details><summary><b>Avatar</b> 5 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar.Active.normal.chromium.png|5| Changed|
|Avatar.Active Appearance - Dark Mode.normal.chromium.png|2| Changed|
|Avatar.Active Appearance.normal.chromium.png|10| Changed|
|Avatar.Color.normal.chromium.png|1| Changed|
|Avatar.Color - RTL.normal.chromium.png|1| Changed|

</details>


  <details><summary><b>Badge</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Badge.Appearance - Dark Mode.normal.chromium.png|15| Changed|
|Badge.Appearance.normal.chromium.png|1| Changed|
|Badge.Default.normal.chromium.png|1| Changed|
|Badge.Size.normal.chromium.png|6| Changed|
|Badge.Shape.normal.chromium.png|1| Changed|
|Badge.Color.normal.chromium.png|1| Changed|

</details>


  <details><summary><b>Button</b> 15 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Button.Large Icon Only - Dark Mode.pressed.chromium.png|5| Changed|
|Button.Large Icon Only - Dark Mode.hover.chromium.png|3| Changed|
|Button.Large Icon Only.default.chromium.png|5| Changed|
|Button.Large With Icon - RTL.hover.chromium.png|4| Changed|
|Button.Large Icon Only.pressed.chromium.png|3| Changed|
|Button.Large With Icon - Dark Mode.hover.chromium.png|3| Changed|
|Button.Large With Icon.hover.chromium.png|4| Changed|
|Button.Large With Icon - RTL.default.chromium.png|5| Changed|
|Button.Large With Icon - Dark Mode.pressed.chromium.png|5| Changed|
|Button.Large Icon Only - Dark Mode.default.chromium.png|4| Changed|
|Button.Large With Icon.default.chromium.png|5| Changed|
|Button.Large With Icon - Dark Mode.default.chromium.png|4| Changed|
|Button.Large Icon Only.hover.chromium.png|4| Changed|
|Button.Large With Icon - RTL.pressed.chromium.png|3| Changed|
|Button.Large With Icon.pressed.chromium.png|3| Changed|

</details>


  <details><summary><b>Divider</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Divider.With Svg - RTL.normal.chromium.png|3| Changed|
|Divider.With Svg.normal.chromium.png|3| Changed|
|Divider.Vertical With Svg.normal.chromium.png|4| Changed|

</details>


  <details><summary><b>MenuList</b> 16 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|MenuList.Checkbox With Icons - RTL.1st selected.chromium.png|4| Changed|
|MenuList.Checkbox With Icons - RTL.normal.chromium.png|4| Changed|
|MenuList.Checkbox With Icons.2nd selected.chromium.png|5| Changed|
|MenuList.Radio With Icons - RTL.1st selected.chromium.png|4| Changed|
|MenuList.Checkbox With Icons - RTL.2nd selected.chromium.png|5| Changed|
|MenuList.Radio With Icons.normal.chromium.png|4| Changed|
|MenuList.Checkbox.2nd selected.chromium.png|1| Changed|
|MenuList.Radio With Icons.2nd selected.chromium.png|4| Changed|
|MenuList.Radio With Icons - RTL.normal.chromium.png|4| Changed|
|MenuList.With Icons.normal.chromium.png|4| Changed|
|MenuList.Checkbox - RTL.2nd selected.chromium.png|1| Changed|
|MenuList.Radio With Icons.1st selected.chromium.png|4| Changed|
|MenuList.Checkbox With Icons.normal.chromium.png|4| Changed|
|MenuList.Radio With Icons - RTL.2nd selected.chromium.png|4| Changed|
|MenuList.Checkbox With Icons.1st selected.chromium.png|4| Changed|
|MenuList.With Icons.hover menuitem.chromium.png|3| Changed|

</details>


  <details><summary><b>Slider</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Slider.Size Small.normal.chromium.png|1| Changed|
|Slider.Size Small.rightArrow.chromium.png|1| Changed|

</details>


  <details><summary><b>Switch</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Switch.Checked - RTL.hover.chromium.png|1| Changed|
|Switch.Checked.hover.chromium.png|1| Changed|

</details>


  <details><summary><b>Text</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Text.Block.normal.chromium.png|995| Changed|

</details>


  <details><summary><b>TextInput</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|TextInput.Size Large.normal.chromium.png|2| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Review by chrisdholt — APPROVED** [review]

**Review by davatron5000 — APPROVED** [review]

**Pull request merged**